public_html/src/appealObject.php

```diff
@@ -1,894 +1,896 @@
+   /**
+    * Status of the appeal
+    */
+   protected $status;
+   
+   /**
+    * User agent
+    */
+   protected $useragent = false;
+
+   /**
+    * Email verification token
+    */
+   protected $emailToken;
+
+   // Maps from DB columns to object fields.
+   private static $columnMap = array(
+      'appealID'     => 'appealID',
+      'email'        => 'emailAddress',
+      'ip'        => 'ipAddress',
+      'wikiAccountName' => 'accountName',
+      'autoblock'    => 'isAutoBlock',
+      'hasAccount'      => 'hasAccount',
+      'blockingAdmin'      => 'blockingAdmin',
+      'timestamp'    => 'timestamp',
+      'appealText'      => 'appeal',
+      'intendedEdits'      => 'intendedEdits',
+      'blockReason'     => 'blockReason',
+      'otherInfo'    => 'otherInfo',
+      'status'    => 'status',
+      'handlingAdmin'      => 'handlingAdmin',
+      'oldHandlingAdmin'   => 'oldHandlingAdmin',
+      'lastLogId'    => 'lastLogId',
+      'emailToken'      => 'emailToken');
+
+   private static $badAccountCharacters = '# / | [ ] { } < > @ % : $';
+     
+   private static $config = "UTRSBot";
+
+   public static function getColumnMap() {
+      return self::$columnMap;
+   }
+
+   public static function getColumnsForSelect($table_alias = 'appeal') {
+      return parent::getColumnsForSelectBase(array_keys(self::$columnMap), 'appeal_', $table_alias);
+   }
+   
+   /**
+    * Build a Appeal object. If $fromDB is true, the mappings in $values
+    * will be assumed to be those from the database; additionally,
+    * the values will not be validated and the object will not be inserted
+    * the DB. Otherwise, the mappings in $values will be assumed to be those
+    * from the appeals form; values will be validated, and the object
+    * will be inserted into the DB on completion.
+    * 
+    * @param array $values the information to include in this appeal
+    * @param boolean $fromDB is this from the database?
+    */
+   private function __construct($values = false){
+      debug('In constuctor for Appeal <br/>');
+
+      if (is_array($values)) {
+         $this->populate($values);
+      }
+
+      debug('Exiting constuctor <br/>');
+   }
+
+   public static function newUntrusted($values) {
+	   
+      $objPeachy = Peachy::newWiki( self::$config );
+	  
+      $appeal = new Appeal($values);
+
+      $appeal->ipAddress = self::getIPFromServer();
+      $appeal->status = self::$STATUS_UNVERIFIED;
+      $appeal->handlingAdmin = null;
+      $appeal->handlingAdminObject = null;
+	  
+	  //Get blocking admin from API
+	  // WARNING: These need to be the raw values to get the appropriate block information from
+	  // the API. DO NOT CHANGE.
+      if ($appeal->isAutoBlock()) {
+	  	$blockinfo = $objPeachy->initUser( $appeal->getIP() )->get_blockinfo();
+      }
+      else {
+      	$blockinfo = $objPeachy->initUser( $appeal->getCommonName() )->get_blockinfo();
+      }
+	if (array_key_exists("by",$blockinfo)) {
```

Comment:
Just to be clear, this if statement is the only thing that actually changed in this code.